### PR TITLE
✨FE: Expiration date

### PR DESCRIPTION
**Pull request by odeimaiz**
## What do these changes do?
- Alert users when the account expires in less than 7 days
- Show expiration date in Profile Page

### Bonus:
- Atomic calls for updating profile
- UserMenuButton bound to first name
- Bugfix: ```Unknown Label``` in files tree are now known

expirationDate

## Related issue/s

- Trial account: related to https://github.com/ITISFoundation/osparc-issues/issues/670

### CODE REVIEW

File: services/web/client/source/class/osparc/Application.js [hunk 1]
```diff
@@ -350,12 +350,34 @@ qx.Class.define("osparc.Application", {
...
+      osparc.data.Resources.getOne("profile")
+        .then(profile => {
+          if ("expirationDate" in profile) {
+            const now = new Date();
+            const daysToExpiration = osparc.utils.Utils.daysBetween(now, new Date(profile["expirationDate"]));
+            console.log("daysToExpiration", daysToExpiration);
+            if (daysToExpiration < 1) {
+              // Will not get here. Backend should not allow
+              let msg = this.tr("This account is expired.<br>");
+              msg += this.tr("Please, contact us at email:<br>");
```
Comment (elisabettai) [inline services/web/client/source/class/osparc/Application.js:362]: Maybe "Please, contact us *by* email" sounds better...

File: services/web/client/source/class/osparc/Application.js [hunk 2]
```diff
@@ -350,12 +350,34 @@ qx.Class.define("osparc.Application", {
...
+              osparc.store.StaticInfo.getInstance().getSupportEmail()
+                .then(supportEmail => osparc.component.message.FlashMessenger.getInstance().logAs(msg+supportEmail, "ERROR"));
+              this.logout();
+              return;
+            } else if (daysToExpiration < 7) {
+              let msg = this.tr("This account will expire in ") + daysToExpiration + this.tr(" days<br>");
+              msg += this.tr("Please, contact us at email:<br>");
```
Comment (elisabettai) [inline services/web/client/source/class/osparc/Application.js:369]: Maybe "Please, contact us *by* email" sounds better...

File: services/web/client/source/class/osparc/desktop/preferences/pages/ProfilePage.js [hunk 3]
```diff
@@ -87,6 +87,20 @@ qx.Class.define("osparc.desktop.preferences.pages.ProfilePage", {
...
       box.add(new qx.ui.form.renderer.Single(form));
 
+      const expirationLayout = new qx.ui.container.Composite(new qx.ui.layout.HBox(5)).set({
+        paddingLeft: 16,
+        visibility: "excluded"
+      });
+      expirationLayout.add(new qx.ui.basic.Label(this.tr("Expiration date:")));
+      const expirationDate = new qx.ui.basic.Label();
+      expirationLayout.add(expirationDate);
+      const infoLabel = this.tr("Please, contact us at email:<br>");
```
Comment (elisabettai) [inline services/web/client/source/class/osparc/desktop/preferences/pages/ProfilePage.js:97]: Maybe "Please, contact us *by* email" sounds better...

**Review by elisabettai — APPROVED** [review]
Nice!
Minor comments.

File: services/web/client/source/class/osparc/desktop/preferences/pages/ProfilePage.js (hunk 3 above)
Comment (odeimaiz) [reply services/web/client/source/class/osparc/desktop/preferences/pages/ProfilePage.js:97]: Google says 20M vs 1M, you win

**Review by sanderegg — APPROVED** [review]
tremendeous

**Review by pcrespov — APPROVED** [review]

**Pull request merged**